ignite/metrics/epoch_metric.py

```diff
@@ -133,12 +138,39 @@ def compute(self) -> float:
             # Run compute_fn on zero rank only
             result = self.compute_fn(_prediction_tensor, _target_tensor)
 
+        # compute_fn outputs: scalars, tensors, tuple/list/mapping of tensors.
+        if not _is_scalar_or_collection_of_tensor(result):
+            raise TypeError(
+                "output not supported: compute_fn should return scalar, tensor, tuple/list/mapping of tensors"
+            )
+
         if ws > 1:
             # broadcast result to all processes
-            result = cast(float, idist.broadcast(result, src=0))
+            return apply_to_type(  # type: ignore
+                result,
+                (torch.Tensor, float, int),  # type: ignore
```

Comment:
Yeah, that's true, @Moh-Yakoub please, update `apply_to_type`  and remove type ignore from here. Thanks !

Reply:
I thought we agreed it would be simpler to keep the method simple and just apply ignore for the type https://github.com/pytorch/ignite/pull/1758#discussion_r593812175

Reply:
I speaking about the second `# type: ignore`.